File: 0101-symmetric-tree/0101-symmetric-tree.cpp
class Solution {
public:
    bool is(TreeNode *l,TreeNode *r){
        if(l==nullptr && r==nullptr)return true;
        if(l==nullptr || r==nullptr)return false;
        if(l->val==r->val){
            return is(l->left,r->right)&&is(l->right,r->left);
        }else{
            return false;
        }
    }
    bool isSymmetric(TreeNode* root) {
        if(root==nullptr)return true;
        return is(root->left,root->right);
    }
};

File: 0101-symmetric-tree/0101-symmetric-tree.py
class Solution(object):
    def isM(self,l,r):
        if l is None and r is None:
            return True
        if l is None or r is None:
            return False
        if l.val==r.val:
            return self.isM(l.left,r.right) and self.isM(l.right,r.left)
        else:
            return False
    def isSymmetric(self, root):
        if root is None:
            return True
        return self.isM(root.left,root.right)